From ec7a5ea27ef759c67225b1616e4266bcff0d7fb4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20=C4=8Cerm=C3=A1k?= <sairon@sairon.cz>
Date: Tue, 14 Jan 2025 18:27:20 +0100
Subject: [PATCH] dts: bcm2712-rpi: Disable CQE for CM5 eMMC to work around FS
 lockups
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

With CQE enabled, HAOS first boot often fails during swap file initilization,
and leaves hanging on all I/O access. For some reason, this only happens when
an ext4lazyinit on the accessed partition hasn't been finished yet. Before the
real solution is found, disable CQE at the cost of a small performance cost.

Signed-off-by: Jan Čermák <sairon@sairon.cz>
---
 arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
index 099d14ecf19a4..d49fc876cccc1 100644
--- a/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm64/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -339,7 +339,7 @@ &sdio1 {
 	mmc-hs400-1_8v;
 	mmc-hs400-enhanced-strobe;
 	broken-cd;
-	supports-cqe = <1>;
+	supports-cqe = <0>;
 	status = "okay";
 };
 
